### PR TITLE
@types/marko - Simplify/improve `marko/express` definitions

### PR DESCRIPTION
**Pull request by darkwebdev**
These changes use the `open interface` pattern of TS defs. This implementation is much easier to use and more correct.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/marko/express.d.ts [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-    interface Application extends express.Application {
-        get: ((name: string) => any) & ExpressRouterMatcher<this> & MarkoRouterMatcher;
-    }
-
-    type MarkoRouterMatcher = (path: PathParams, ...handlers: RequestHandler[]) => Application;
-    type RequestHandler = (req: express.Request, res: Response, next?: NextFunction) => any;
 }
 
-export = m;
+export = express;
```
Comment (PranavSenthilnathan) [inline types/marko/express.d.ts]: Is this line needed? As far as I understand, this definition is a module augmentation on `express-serve-static-core` and no functions/values are exported directly.

Comment (darkwebdev) [reply types/marko/express.d.ts]: According to the (Marko.js docs)[https://markojs.com/docs/express/] we need to do this:
```
var express = require("express");
var markoExpress = require("marko/express")

var app = express();
app.use(markoExpress()); // <- this is the reason I export `express`
```

I have this example in the tests and I don't know how to make it better, do you have any ideas?

Comment (PranavSenthilnathan) [reply types/marko/express.d.ts]: Oops, I didn't read at the docs carefully enough. Your change looks correct.

**Comment by redonkulus** [comment]
@darkwebdev @PranavSenthilnathan I just updated my app to version 4.6.2 (from 4.6.1) and I get the following error from tsc:

```
14:48:42   src/app.ts:27:9 - error TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'typeof e' has no compatible call signatures.
14:48:42  
14:48:42   27 app.use(markoExpress());
14:48:42              ~~~~~~~~~~~~~~
14:48:42  
14:48:42     node_modules/@types/marko/express.d.ts:1:1
14:48:42       1 import * as express from 'express';
14:48:42         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
14:48:42       Type originates at this import. A namespace-style import cannot be called or constructed, and will cause a failure at runtime. Consider using a default import or import require here instead.
```

My app is just doing:

```
import markoExpress from 'marko/express';
app.use(markoExpress());
```

Info:
- typescript@3.5.3
- marko@4.18.11
- @types/marko@4.6.2

**Comment by darkwebdev** [comment]
@redonkulus try `import * as markoExpress from 'marko/express';`
as far as I know that's the default for modules exported using old-school commonJS style like `marko/express` which does:

```
module.exports = function markoAppMiddleware() {
...
```

**Comment by redonkulus** [comment]
Thanks @darkwebdev. Your suggestion wasn't working for me, but `import { default as markoExpress } from 'marko/express';` seems to work. 

I'm surprised this isn't breaking anyone else.

**Comment by darkwebdev** [comment]
@redonkulus looks like usage depends on your `tsconfig`:
- if you have `"allowSyntheticDefaultImports": true` or `"esModuleInterop": true` then use `import { default as ...}`
- otherwise use `import * as ...`

**Pull request merged**